src/main/java/com/softserve/demo/repository/CustomerRepository.java

```diff
@@ -1,11 +1,13 @@
 package com.softserve.demo.repository;
 
 import com.softserve.demo.model.Customer;
+import com.softserve.demo.model.Offer;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.stereotype.Repository;
 
 @Repository
 public interface CustomerRepository extends JpaRepository<Customer, Integer> {
     boolean existsByEmail(String email);
     Customer findByEmail(String email);
+    Customer getCustomerByOffers(Offer offer);
```

Comment:
by offer ? you are passing single offer so method should be getCustomerByOffer

Reply:
java.lang.IllegalArgumentException: Failed to create query for method public abstract com.softserve.demo.model.Customer com.softserve.demo.repository.CustomerRepository.getCustomerByOffer(com.softserve.demo.model.Offer)! No property offer found for type Customer! Did you mean 'offers'?

Reply:
I get it!